src/VisualStudio/Core/Def/Implementation/VisualStudioWorkspaceContextService.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    [ExportWorkspaceService(typeof(IWorkspaceContextService), ServiceLayer.Host), Shared]
+    internal sealed class VisualStudioWorkspaceContextService : IWorkspaceContextService
+    {
+        /// <summary>
+        /// Guid for UI context set by liveshare upon joining as a client to a session.
+        /// </summary>
+        private static readonly Guid s_sessionJoinedUIContextGuid = Guid.Parse("c6f0e3cb-a3c3-49bd-bad2-7aad8690c15b");
+        private static readonly UIContext s_sessionJoinedUIContext = UIContext.FromUIContextGuid(s_sessionJoinedUIContextGuid);
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public VisualStudioWorkspaceContextService()
+        {
+        }
+
+        public bool IsInRemoteClientContext()
```

Comment:
Is this going to answer for all workspaces? What about things like interactive window that are still running locally, even in the remote case?

Reply:
@jasonmalinowski is it possible to have this be exported only for misc files / VS workspace?  It's currently exported for the Host layer but I admit I don't fully understand which workspaces that applies to.  

I guess my idea would be that unless otherwise specified, workspaces like the interactive window get the default service.

Reply:
@dibarbet: so some workspaces have their own special "kinds" that you can export on; but alas the main VisualStudioWorkspace does not. Your usual option here is to either export a workspace service _factory_ that can type check the workspace type (with `workspace is VisualStudioWorkspace`) or have this method here take the Workspace as a parameter and you can check that way.

Reply:
Ahh the workspace service factory seems like a good solution that way I can just check once.  I'll try it out, thanks!